### ISSUE TITLE
Backticks used in JSDoc in a particular way cause tsserver to crash

### ISSUE THREAD

**Issue by westonpace** [microsoft/TypeScript#31868]
**TypeScript Version:**  3.5.1

**Search Terms:** JSDoc backticks tsserver crash no intellisense

**Code**

```ts
/**
* @param `arg` str: string argument that will be shown in message
*/
function hello(arg: string) {
    console.log(arg);
}

hello('World');
```

**Expected behavior:**
This compiles fine with `tsc`.  When loaded into VSCode however tsserver crashes without any meaningful error message.

**Actual behavior:**
tsserver should not crash. An error would be fine if this is invalid JSDoc (I don't actually know JSDoc).

**Playground Link:** 
Since this only happens in tsserver I have no playground link

**Related Issues:** 
I believe this may be related to this change: https://github.com/microsoft/TypeScript/commit/c3a9429420452fdfad73e9d947e52ca825a4b8f7

The issue is not present in 3.4

The function `parseBracketNameInPropertyAndParamTag` calls `parseExpectedTokenJSDoc(SyntaxKind.BacktickToken);` and `parseExpectedTokenJSDoc` then calls `ts.tokenToString(t)` on `SyntaxKind.BacktickToken` and this returns `undefined`.  This `undefined` then blows up in an assert later on.

**Comment by DanielRX** [comment]
Looking at the linked change I see `// a markdown-quoted name: `arg` is not legal jsdoc, but occurs in the wild`. Doesn't this imply the JSDoc is invalid?

**Comment by rbuckton** [comment]
It looks like this was fixed by #32429.
